# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.cpp
# Commit: 2fc9a3d0ad8b
# Full Hash: 2fc9a3d0ad8bdbce325df78fcd731e925245afaf
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r 2d598be5ec67 -r 2fc9a3d0ad8b dom/cookiestore/CookieChangeEvent.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/cookiestore/CookieChangeEvent.cpp	Wed Sep 11 16:26:54 2024 +0000
@@ -0,0 +1,53 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/dom/CookieChangeEvent.h"
+#include "mozilla/dom/CookieChangeEventBinding.h"
+
+namespace mozilla::dom {
+
+CookieChangeEvent::CookieChangeEvent(EventTarget* aOwner,
+                                     nsPresContext* aPresContext,
+                                     WidgetEvent* aEvent)
+    : Event(aOwner, aPresContext, aEvent) {}
+
+CookieChangeEvent::~CookieChangeEvent() = default;
+
+JSObject* CookieChangeEvent::WrapObjectInternal(
+    JSContext* aCx, JS::Handle<JSObject*> aGivenProto) {
+  return mozilla::dom::CookieChangeEvent_Binding::Wrap(aCx, this, aGivenProto);
+}
+
+void CookieChangeEvent::GetChanged(nsTArray<CookieListItem>& aList) const {
+  aList = mChanged.Clone();
+}
+
+void CookieChangeEvent::GetDeleted(nsTArray<CookieListItem>& aList) const {
+  aList = mDeleted.Clone();
+}
+
+/* static */ already_AddRefed<CookieChangeEvent> CookieChangeEvent::Constructor(
+    const GlobalObject& aGlobal, const nsAString& aType,
+    const CookieChangeEventInit& aEventInit) {
+  nsCOMPtr<EventTarget> t = do_QueryInterface(aGlobal.GetAsSupports());
+  RefPtr<CookieChangeEvent> event = new CookieChangeEvent(t, nullptr, nullptr);
+  bool trusted = event->Init(t);
+
+  event->InitEvent(aType, aEventInit.mBubbles, aEventInit.mCancelable);
+  event->SetTrusted(trusted);
+
+  if (aEventInit.mChanged.WasPassed()) {
+    event->mChanged = aEventInit.mChanged.Value();
+  }
+
+  if (aEventInit.mDeleted.WasPassed()) {
+    event->mDeleted = aEventInit.mDeleted.Value();
+  }
+
+  return event.forget();
+}
+
+}  // namespace mozilla::dom